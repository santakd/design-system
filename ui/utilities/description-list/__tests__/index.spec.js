/* eslint-env jest */
import React from 'react';
import createHelpers from '../../../../jest.helpers';
const { matchesMarkup } = createHelpers(__dirname);

describe('renders description list utilities', () => {
  it('renders description list', () =>
    matchesMarkup(
      <dl>
        <dt>First Label:</dt>
        <dd>Description for first label</dd>
        <dt>Second Label:</dt>
        <dd>Description for second label</dd>
      </dl>
    ));

  it('renders description list with inline name/value pair', () =>
    matchesMarkup(
      <dl className="slds-dl_inline">
        <dt className="slds-dl_inline__label">First Label:</dt>
        <dd className="slds-dl_inline__detail">Description for first label</dd>
        <dt className="slds-dl_inline__label">Second Label:</dt>
        <dd className="slds-dl_inline__detail">Description for second label</dd>
      </dl>
    ));

  it('renders description list with horizontal name/value pair', () =>
    matchesMarkup(
      <dl className="slds-dl_horizontal">
        <dt className="slds-dl_horizontal__label">First Label:</dt>
        <dd className="slds-dl_horizontal__detail">
          Description for first label
        </dd>
        <dt className="slds-dl_horizontal__label">Second Label:</dt>
        <dd className="slds-dl_horizontal__detail">Second description</dd>
      </dl>
    ));

  it('renders description list with inline name/value pair in narrow region', () =>
    matchesMarkup(
      <div className="slds-region_narrow" style={{ width: '360px' }}>
        <dl className="slds-dl_inline">
          <dt className="slds-dl_inline__label">First Label:</dt>
          <dd className="slds-dl_inline__detail">
            Description for first label
          </dd>
          <dt className="slds-dl_inline__label">Second Label:</dt>
          <dd className="slds-dl_inline__detail">Second description</dd>
        </dl>
      </div>
    ));

  it('renders description list with horizontal name/value pair in narrow region', () =>
    matchesMarkup(
      <div className="slds-region_narrow" style={{ width: '360px' }}>
        <dl className="slds-dl_horizontal">
          <dt className="slds-dl_horizontal__label">First Label:</dt>
          <dd className="slds-dl_horizontal__detail">
            Description for first label
          </dd>
          <dt className="slds-dl_horizontal__label">Second Label:</dt>
          <dd className="slds-dl_horizontal__detail">Second description</dd>
        </dl>
      </div>
    ));
});
