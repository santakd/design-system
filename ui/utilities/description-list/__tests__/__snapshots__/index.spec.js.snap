// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders description list utilities renders description list 1`] = `
<dl>
  <dt>
    First Label:
  </dt>
  <dd>
    Description for first label
  </dd>
  <dt>
    Second Label:
  </dt>
  <dd>
    Description for second label
  </dd>
</dl>
`;

exports[`renders description list utilities renders description list with horizontal name/value pair 1`] = `
<dl
  className="slds-dl_horizontal"
>
  <dt
    className="slds-dl_horizontal__label"
  >
    First Label:
  </dt>
  <dd
    className="slds-dl_horizontal__detail"
  >
    Description for first label
  </dd>
  <dt
    className="slds-dl_horizontal__label"
  >
    Second Label:
  </dt>
  <dd
    className="slds-dl_horizontal__detail"
  >
    Second description
  </dd>
</dl>
`;

exports[`renders description list utilities renders description list with horizontal name/value pair in narrow region 1`] = `
<div
  className="slds-region_narrow"
  style={
    Object {
      "width": "360px",
    }
  }
>
  <dl
    className="slds-dl_horizontal"
  >
    <dt
      className="slds-dl_horizontal__label"
    >
      First Label:
    </dt>
    <dd
      className="slds-dl_horizontal__detail"
    >
      Description for first label
    </dd>
    <dt
      className="slds-dl_horizontal__label"
    >
      Second Label:
    </dt>
    <dd
      className="slds-dl_horizontal__detail"
    >
      Second description
    </dd>
  </dl>
</div>
`;

exports[`renders description list utilities renders description list with inline name/value pair 1`] = `
<dl
  className="slds-dl_inline"
>
  <dt
    className="slds-dl_inline__label"
  >
    First Label:
  </dt>
  <dd
    className="slds-dl_inline__detail"
  >
    Description for first label
  </dd>
  <dt
    className="slds-dl_inline__label"
  >
    Second Label:
  </dt>
  <dd
    className="slds-dl_inline__detail"
  >
    Description for second label
  </dd>
</dl>
`;

exports[`renders description list utilities renders description list with inline name/value pair in narrow region 1`] = `
<div
  className="slds-region_narrow"
  style={
    Object {
      "width": "360px",
    }
  }
>
  <dl
    className="slds-dl_inline"
  >
    <dt
      className="slds-dl_inline__label"
    >
      First Label:
    </dt>
    <dd
      className="slds-dl_inline__detail"
    >
      Description for first label
    </dd>
    <dt
      className="slds-dl_inline__label"
    >
      Second Label:
    </dt>
    <dd
      className="slds-dl_inline__detail"
    >
      Second description
    </dd>
  </dl>
</div>
`;
